avogadro/core/cube.h

```diff
@@ -235,6 +239,45 @@ class AVOGADROCORE_EXPORT Cube
    */
   Mutex* lock() const { return m_lock; }
 
+  /**
+   * Compute the gradient at a specific point in the cube.
+   * @param i x index
+   * @param j y index
+   * @param k z index
+   * @return Gradient vector at the specified point.
+   */
+  std::array<float, 3> computeGradient(int i, int j, int k) const;
+
+  /**
+   * Get the values of the eight corners of a cube cell.
```

Comment:
I'm not quite sure I understand from the docs here. I've got some index (i, j, k) into the cube. What's the "cell" of the cube? Is it just going to get me the corners of the entire cube? If so, why do I need i, j, k parameters?